hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ScmTableCountTask.java

```diff
@@ -0,0 +1,212 @@
+  private static final long NANOSECONDS_IN_MILLISECOND = 1_000_000;
+
+  public ScmTableCountTask(ReconStorageContainerManagerFacade reconSCM,
+                           ReconTaskStatusDao reconTaskStatusDao,
+                           ReconTaskConfig reconTaskConfig,
+                           ScmTableCountDao scmTableCountDao,
+                           UtilizationSchemaDefinition schemaDefinition) {
+    super(reconTaskStatusDao);
+    this.scmDBStore = reconSCM.getScmDBStore();
+    this.scmTableCountDao = scmTableCountDao;
+    this.dslContext = schemaDefinition.getDSLContext();
+    this.interval = reconTaskConfig.getScmTableCountTaskInterval().toMillis();
+  }
+
+  @Override
+  protected synchronized void run() {
+    try {
+      while (canRun()) {
+        wait(interval);
+        runTableCountProcess();
+      }
+    } catch (Throwable t) {
+      LOG.error("Error while running ScmTableCountTask: {}", t);
+      if (t instanceof InterruptedException) {
+        Thread.currentThread().interrupt();
+      }
+    }
+  }
+
+  private void runTableCountProcess() {
+      try {
+        long startTime, endTime, duration, durationMilliseconds;
+
+        int execute = dslContext.truncate(SCM_TABLE_COUNT_TABLE_NAME).execute();
+        LOG.info("Deleted {} records from {}", execute,
+            SCM_TABLE_COUNT_TABLE_NAME);
+
+        startTime = System.nanoTime();
+        processTableCount();
+        endTime = System.nanoTime();
+        duration = endTime - startTime;
+        durationMilliseconds = duration / NANOSECONDS_IN_MILLISECOND;
+        LOG.info(
+            "Elapsed Time in milliseconds for processTableCount() execution: {}",
+            durationMilliseconds);
+      } catch (Exception e) {
+        LOG.error("An error occurred while performing table count: {}", e);
+      }
+    }
+
+  /**
+   * Processes the table count by iterating over SCM tables and retrieving the
+   * counts of objects in each table. The counts are then stored in the Recon
+   * database.
+   *
+   * @throws IOException if an I/O error occurs during table count processing.
+   */
+  public void processTableCount() throws IOException {
+    // Acquire write lock
+    lock.writeLock().lock();
+    try {
+      // Initialize the object count map
+      Map<String, Long> objectCountMap = initializeCountMap();
+
+      // Iterate over SCM tables
+      for (String tableName : getTaskTables()) {
+        Table table = scmDBStore.getTable(tableName);
+
+        try (TableIterator keyIter = table.iterator()) {
+          // Retrieve the count of objects in the table
+          long count = Iterators.size(keyIter);
+          objectCountMap.put(getRowKeyFromTable(tableName), count);
+        } catch (IOException ioEx) {
+          LOG.error("Unable to populate SCM Table Count in Recon DB.", ioEx);
+        }
+      }
+
+      // Write the counts to the Recon db if the object count map is not empty
+      if (!objectCountMap.isEmpty()) {
```

Comment:
Pls re-check this logic

Reply:
I believe We really do not require the following check
```
      // Write the counts to the Recon db if the object count map is not empty
      if (!objectCountMap.isEmpty()) {
        writeCountsToDB(objectCountMap);
      }
```
The reason for this is that the table will be truncated at the start of the task during every time interval. Therefore, even if the initialized values do not change, it is still necessary to update the table with some data. This is because the table will be read by the APIs, and if it remains empty, there is a possibility of encountering a (NPE) when they attempt to access it.

Ill remove the check
Thanks for the comment @devmadhuu